replication.go

```diff
@@ -1842,47 +1852,65 @@ func (cm *ContentManager) retrieveContent(ctx context.Context, contentToFetch ui
 		close(prog.wait)
 	}()
 
-	if err := cm.runRetrieval(ctx, contentToFetch); err != nil {
+	contentRoot, err := cm.runRetrieval(ctx, contentToFetch)
+	if err != nil {
 		prog.endErr = err
 		return err
 	}
 
+	// FIXME need to do something with this
+	panic(contentRoot)
```

Comment:
eh?

Reply:
When you do a partial retrieval - you get a new CID for your sub-dag. The client doesn't know what it is, just the path to it. The code pulls it out for you by re-running the selector, and then you do... something with it.

Reply:
@whyrusleeping an alternative would be to run the selector-over-your BS lazily, when the user requests their data back. Then changing the functions args won't be needed.

It is currently done this way to minimize the amount of time for GC ( since you can't pin the partial dag )...

Dunno - thoughts welcome.